narwhal/primary/src/primary.rs

```diff
@@ -69,6 +69,9 @@ pub mod primary_tests;
 /// The default channel capacity for each channel of the primary.
 pub const CHANNEL_CAPACITY: usize = 1_000;
 
+/// The number of shutdown receivers to create on startup. We need one per component loop.
+pub const NUM_SHUTDOWN_RECEIVERS: u64 = 25;
```

Comment:
I've defined this to a default high value so we don't need to keep updating it, since having extra subscribed receivers is inconsequential, but having not enough will lead to a panic (the panic is a tradeoff is a race condition). I am not completely satisfied with this and would rather not have to define this at all. Any suggestions of a way to avoid needing to define this are welcome.